Repository/RevisionRepository.php

```diff
@@ -418,6 +418,34 @@ public function findByOuuidAndContentTypeAndEnvironnement(ContentType $contentTy
         return $out[0];
     }
 
+    /**
+     * @param string $ouuid
+     * @param int $contentType
+     * @param int $env
+     * @return null|Revision
+     * @throws NonUniqueResultException
+     */
+    public function findIdByOuuidAndContentTypeAndEnvironment(string $ouuid, int $contentType, int $env)
+    {
+        $qb = $this->createQueryBuilder('r');
+        $qb->join('r.environments', 'e');
+        $qb->where('r.ouuid = :ouuid and e.id = :envId and r.contentType = :contentTypeId');
+        $qb->setParameters([
+            'ouuid' => $ouuid,
+            'envId' => $env,
+            'contentTypeId' => $contentType
+        ]);
+
+        $out = $qb->getQuery()->getResult();
+        if (count($out) > 1) {
+            throw new NonUniqueResultException($ouuid . ' is publish multiple times in ' . $env);
+        }
+        if (empty($out)) {
```

Comment:
return $out[0] ?? null;

Reply:
I also want the if (empty($out) removed. If you use getArrayResult() instead of getResult, doctrine will always return an array. Check the count throw error if bigger then 1 else return $out[0] ?? null;